samcli/cli/types.py

```diff
@@ -194,6 +219,7 @@ def __init__(self, multiple_values_per_key=False):
     TAG_REGEX = '[A-Za-z0-9\\"_:\\.\\/\\+-\\@=]'
 
     _pattern = r"{tag}={tag}".format(tag=_generate_match_regex(match_pattern=TAG_REGEX, delim=" "))
+    _quoted_pattern = _generate_quoted_match_regex(match_pattern=TAG_REGEX)
```

Comment:
Can we reuse or refactor the old `_generate_match_regex`? They share similar return values.

Reply:
They have different regex, hence creating a new function.

Reply:
The difference between the two is the extra regex string for the deliminator right? Can we make the deliminator argument optional, then just append the extra deliminator regex it was passed in? Another option is to make constants